src/sentry/static/sentry/app/views/projectTags.jsx

```diff
@@ -77,36 +65,54 @@ export default class ProjectTags extends AsyncView {
           <PanelBody>
             {isEmpty && (
               <EmptyMessage>
-                {tct('There are no tags, [link:learn to add tags]', {
+                {tct('There are no tags, [link:learn how to add tags]', {
                   link: (
                     <ExternalLink href="https://docs.sentry.io/hosted/learn/context/" />
                   ),
                 })}
               </EmptyMessage>
             )}
 
-            {!isEmpty &&
-              tags.map(({key, canDelete}, idx) => {
-                return (
+            <Access access={['project:write']}>
+              {({hasAccess}) =>
+                !isEmpty &&
+                tags.map(({key, canDelete}, idx) => (
                   <PanelItem p={0} align="center" key={key} className="ref-tag-row">
                     <Box align="flex-end" flex="1" p={2}>
                       <span>{key}</span>
                     </Box>
                     <Flex align="center" p={2}>
-                      {canDelete ? (
-                        this.renderLink(key, canDelete, idx)
-                      ) : (
-                        <Tooltip title={t('This tag cannot be deleted.')}>
-                          <span>{this.renderLink(key, canDelete, idx)}</span>
-                        </Tooltip>
-                      )}
+                      <Tooltip
+                        disabled={canDelete && hasAccess}
```

Comment:
Could these `canDelete && hasAccess` be combined into one variable? You use them both together in three places with almost the same logic as `!canDelete || !hasAccess` can be [de morganed](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) into `!(canDelete && hasAccess)`. You could do something like `showButton = (canDelete && hasAccess)`

Reply:
Yeah.

Honestly I kind of just didn't want to add the extra levels of nesting by turning this `children` render-prop into a full blown `{}` function.

I will though